trieste/acquisition/multi_objective/pareto.py

```diff
@@ -73,6 +79,194 @@ def hypervolume_indicator(self, reference: TensorType) -> TensorType:
         )
         return hypervolume_indicator
 
+    def sample_diverse_subset(
+        self,
+        sample_size: int,
+        allow_repeats: bool = True,
+        bounds_delta_scale_factor: float = 0.2,
+        bounds_min_delta: float = 1e-9,
+    ) -> tuple[TensorType, TensorType]:
+        """
+        Sample a set of diverse points from the Pareto set using
+        Hypervolume Sharpe-Ratio Indicator
+
+        :param sample_size: The number of point to sample from the Pareto front
+        :param allow_repeats: Whether the sample may contain repeats
+        :param bounds_delta_scale_factor: The factor by which to grow the distance
+            between extrema when calculating lower and upper bounds
+        :param bounds_min_delta: The minimum value of the distance between extrema
+        """
+
+        if cp is None:
+            raise ImportError(
+                "Pareto.sample method requires cvxpy, "
+                "this can be installed via `pip install trieste[qhsri]`"
+            )
+
+        front_size, front_dims = self.front.shape
+
+        if (front_size < sample_size) and allow_repeats is False:
+            raise ValueError(
+                f"Tried to sample {sample_size} points from a Pareto"
+                f" set of size {front_size}, please ensure sample size is smaller than"
+                " Pareto set size."
+            )
+
+        lower_bound, upper_bound = self._get_bounds(bounds_delta_scale_factor, bounds_min_delta)
+
+        p = self._calculate_p_matrix(lower_bound, upper_bound)
+
+        # Calculate q matrix
+        p_diag = np.expand_dims(np.diagonal(p), axis=1)
+        q = p - np.dot(p_diag, np.transpose(p_diag))
+
+        x_star = self._find_x_star(q, p)
+
+        if allow_repeats:
+            samples, sample_ids = self._choose_batch_with_repeats(x_star, sample_size)
+        else:
+            samples, sample_ids = self._choose_batch_no_repeats(x_star, sample_size)
+
+        return samples, sample_ids
+
+    def _choose_batch_with_repeats(
+        self, x_star: TensorType, sample_size: int
+    ) -> tuple[TensorType, TensorType]:
+
+        # Calculate number of times each point is sampled
+        n_times_sampled = x_star * float(sample_size)
```

Comment:
is it OK to convert with `float()`? naive question (maybe the answer is yes)

Reply:
Yes, I'm pretty sure that's what python does under the hood if you do `float * int`. It's unnecessary here, not sure why I've got it in, so I'll remove it